.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,50 @@
+name: SonarCloud Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+          submodules: 'recursive'
+      # For building GeoNetwork, JDK8 is necessary, but for running
+      # the SonarQube plubin, JDK11 is necessary.
+      # So, first install JDK 8, build GeoNetwork, then install JDK11
```

Comment:
A good reason to merge #6276 IMO...

Reply:
* Yes. I planned to add documentations later.
* When this PR is merged, there should also be an overview of the main branch available.
* Java 11 would be really cool.

I would merge this PR now, and add documentation later. This will probably not be the last commit concerning SonarCloud anyways. Is it OK for you to merge now?